test/system/trader.js

```diff
@@ -72,36 +78,77 @@ class Trader {
   }
 
   cancelAllOrders() {
-    let self = this;
     return new Promise((resolve) => {
-      self.client.cancelMarketOrders(self.market)
-        .then(() => {
-          console.log('all orders are cancelled');
-          self.logInitialDetails(self.config);
-          process.exit();
-          resolve();
-        })
-        .catch(err => {
-          console.log(err.message);
-        });
+      if(this.config.name === 'trader1') {
+        this.client.cancelMarketOrders(this.market)
+          .then(() => {
+            console.log('all orders are cancelled for ', this.config.name);
+            allConfig.forEach(config => {
+              this.logInitialDetails(config);
+            });
+            process.exit();
+            resolve();
+          })
+          .catch(err => {
+            console.log(err.message);
+          });
+      }
     });
   }
 
   _execute_trading_strategy() {
     let promises = [];
-    let side;
-    for (let i = 0; i < 4; i++) {
-      side = i <= 1 ? 'B' : 'S';
-      promises.push(this.placeOrder(side));
+    if (this.config.name === 'trader1') {
```

Comment:
I see how this is working but it is going to make the system test more complicated as it takes on more scenarios to test.
System test traders are different from the Simulator traders in that they don't really have Strategies to follow, instead they blindly follow a set of deterministic instructions and expect certain outcomes.

The code structure will benefit by moving the Orchestration code (i.e. the code that decides what to do next) outside of the Trader class. The  Trader class will have functions on it like placeOrders, cancelOrders and so on and the scenario functions will orchestrate what the traders do.
so the index.js file will look a little like this:
```javascript
createTraders(config)
  .then(() => runScenario1())
  .then(() => runScenario2())
  .then(() => runScenario3())
  // ...
```